src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php

```diff
@@ -170,7 +170,14 @@ public function onKernelControllerArguments(ControllerArgumentsEvent $event): vo
                 };
             }
 
-            $arguments[$i] = $payload;
+            if ($argument->metadata->isVariadic()) {
+                if (!(is_array($payload) && array_is_list($payload))) {
+                    throw HttpException::fromStatusCode(422);
+                }
+                array_splice($arguments, $i, count($payload), $payload);
+            } else {
+                $arguments[$i] = $payload;
```

Comment:
What about the situation when the argument is `UploadedFile` and the `$payload` is an array of files? I think this would still result in a `500`?